util/reflect_test.go

```diff
@@ -753,7 +766,25 @@ func TestForEachField(t *testing.T) {
 		return
 	}
 
-	printMapKeysIterFunc := func(ni *NodeInfo, in, out interface{}) (errs Errors) {
+	printSchemaAnnotationFieldsIterFunc = func(ni *NodeInfo, in, out interface{}) (errs Errors) {
+		if !IsValueScalar(ni.FieldValue) || IsValueNil(ni.FieldKey) {
+			return
+		}
+
+		fn, err := SchemaPaths(ni.StructField)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if l := len(fn); l != 1 {
+			errs = append(errs, fmt.Errorf("invalid schema path length %d for %v", l, ni.StructField.Name))
+		}
+
+		outs := out.(*string)
```

Comment:
(optional) Probably pointless since this is only test code, but even that might still serve as an example.

Would we be better off with out being a bytes.Buffer instead of doing the string appends?

Applies below too.

Reply:
It certainly could -- I was trying to keep consistency with the test cases that we had for `forEachFieldInternal`. I added a TODO to refactor this at some point.